opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py

```diff
@@ -151,7 +151,7 @@ def should_sample(
         trace_state: "TraceState" = None,
     ) -> "SamplingResult":
         if self._decision is Decision.DROP:
-            return SamplingResult(self._decision)
+            attributes = None
         return SamplingResult(self._decision, attributes, trace_state)
```

Comment:
Do we need to pass in the `trace_state` for dropped spans?

Reply:
Yes as otherwise the trace breaks when the span is propagated further. E.g. `flask` instrumentation extracts span a span where the sampling flag in the trace parent is `false`. The handler in the `flask` app makes an outgoing HTTP request with the `requests` library. The `requests` instrumentation has the `flask` span as parent so the created span will be dropped. The dropped span however is used for injecting/propagation, so it needs to include the `trace_state` of the parent.
`w3c_tracecontext` tests also rely on that the trace_state gets propagated.